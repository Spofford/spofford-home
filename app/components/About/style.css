.about h2, .about h3 {
  margin-bottom:15px;
}

.about p {
  line-height:1.2em;
  margin-bottom:10px;
}

.about blockquote {
  font-size:1.5em;
  font-style: italic;
  padding-left:15px;
}

.about img {
  width:100%;
}

.about .fullHeader .container {
  background-color: #FF9138;
  box-shadow:none;
}

.about > .fullHeader h1, .about > .fullHeader > .container .fa-bars, .about > .fullHeader > .container .menu-label, .about > .fullHeader .search-container.hidden .fa-search, .about .cancelTrigger, .about .cancelTrigger, .about .search-container .link-container span, .about .search-container .link-container span {
   color:white;
 }

.about p {
  line-height:1.8em;
  margin-bottom:40px;
}


.about a:hover {
  color:#f8f8f8;
}

.about .hero {
  background-color: #FF9138;
  height:calc(100% - 65px);
  margin-top:65px;
}

.about .hero h3 {
  color: white;
}

.about .body-container {
  padding-bottom:40px;
}

#reactHubspotForm0 {
  margin:0px auto;
  display:table;
  width:500px;
}

@media (min-width: 700px) {
	.about .hero .copy-container {
    max-width:61.8%;
    padding-left:60px;
  }
}
