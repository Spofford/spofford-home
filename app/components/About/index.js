import React from "react"
import HubspotForm from 'react-hubspot-form'
const ReactMarkdown = require('react-markdown')
import style from "./style.css"
import * as contentful from 'contentful'
// import { default as client } from '../../../src/services/contentful-client.js'
//import { fetchModel } from "../../redux/actions"

import { default as Header } from "../Header"

export class About extends React.Component {

  state = {
    model: {}
  }

  client = contentful.createClient({
    space: 'cahjy08ew1qz',
    accessToken: '37c6ec31a1a6cb3f533f51fa4c4af8fee88e2f910d9879eb79b2d073ae8cc499'
  })


  componentDidMount() {
    this.fetchModel().then(this.setModel);
    //this.props.dispatch(Actions.fetchModel('wKhyV2HRBeSgK4UcOoiEi'))
    /*
    const response = await client().getEntry('wKhyV2HRBeSgK4UcOoiEi')
    self.setState({
      model: response.fields
    })
    */
    //const client = await contentful.createClient({
    //  space: 'cahjy08ew1qz',
    //  accessToken: '37c6ec31a1a6cb3f533f51fa4c4af8fee88e2f910d9879eb79b2d073ae8cc499'
    // })
    //const model = await client.getEntry('wKhyV2HRBeSgK4UcOoiEi')

    //this.setModel;

    // var script = document.querySelector('[src="https://js.hsforms.net/forms/v2.js"]')

  	//if(window.hbspt) {
    //	window.hbspt.forms.create({
    //  	portalId: '',
    //    formId: '',
    //    target: '#hubspotForm'
    //  })
    // }
  }

  fetchModel = () => this.client.getEntry('wKhyV2HRBeSgK4UcOoiEi')

  setModel = response => {
    this.setState({
      model: response.fields
    })
  }

  render() {
    return (
      <div className="about">
        <Header headerStart={true} />
        <div className="hero">
          <div className="copy-container">
            <h2>{this.state.model.head}</h2>
            <h3>{this.state.model.subhead}</h3>
          </div>
        </div>
        <div className="body-container">
          <div className="copy-container">

            <ReactMarkdown source={this.state.model.bodyText} />
            <HubspotForm
               portalId='4042167'
               formId='d1e13228-c396-4338-bbe3-67cd84f53065'
               />
          </div>
        </div>
      </div>
    )
  }
}

export default About
