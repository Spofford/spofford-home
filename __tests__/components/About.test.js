import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import 'jest-dom/extend-expect';
import { createClient } from "contentful";

import About from '../../app/components/About';

const client = createClient({
  space: 'foo-space',
  accessToken: 'bar-token',
  host: "127.0.0.1:8090",
  insecure: true
});

/*
const mockStore = configureStore();
const initialState = {
  user: {
    email: "amhasler@gmail.com",
    first_name: "Adam",
    last_name: "Hasler",
    role: 1,
    id: 20,
    charges: []
  }
}
const store = mockStore(initialState);

const contentModel = {
  fields: {
    head: "radically rewriting the book",
    subhead: "we're going to change everything",
    bodyText: "Content content content",
    visual: {
      sys: {
        id: 10
      }
    },
    subhead2: "more content",
    bodyText2: "even more content",
    visual: {
      sys: {
        id: 12
      }
    },
    subhead3: "more content",
    bodyText3: "even more content",
    visual: {
      sys: {
        id: 13
      }
    },
    subhead3: "more content",
    bodyText3: "even more content",
  }
}
*/

/*
describe('Contentful mocking', () => {
  it('should be possible to mock Contentful', (done) => {
    const client = { getEntry: () => { return Promise.resolve(); } };
    const spy = {
      fn: (value) => {
        expect(value).toBe('yeah');
        done();
      },
    };

    spyOn(contentful.default, 'createClient').and.returnValue(client);

    fetchModel()(spy.fn);
  });
});
*/
describe('<About />', () => {
  describe('render()', () => {
    // jest.mock("./__mocks__/contentful.js")

    //test('renders the component', () => {
    //  const wrapper = shallow(<About />);
      // const component = wrapper.dive();

      //expect(toJson(wrapper)).toMatchSnapshot();
    //});

    //test('renders Hubspot form', () => {
    //  const wrapper = mount(<div className="hbspt-form" />);
    //  expect(wrapper.exists('.hbspt-form')).toEqual(true)

    // })

    //test('calls setModel on mount', () => {
      //const wrapper = shallow(<About />);

      //setTimeout(() => {
      //  wrapper.update();

      //  const state = wrapper.instance().state;

      //  console.log(state)
        //expect(state.term).toEqual("Mountains");
        //expect(state.status).toEqual("done");
        //expect(state.images.length).toEqual(1);

        //expect(wrapper.find("Image").length).toEqual(1);

      //  done();
      // });

      /*
      const wrapper = shallow(<About />);
      const spy = jest.spyOn(wrapper.instance(), "setModel");
      wrapper.instance().forceUpdate();
      const data = await wrapper.instance().fetchModel();
      expect(spy).toHaveBeenCalled();
      done()
      */
    //})
  });
});
