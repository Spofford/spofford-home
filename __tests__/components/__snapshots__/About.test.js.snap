// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> render() renders the component 1`] = `
<div
  className="about"
>
  <Connect(Header)
    headerStart={true}
  />
  <div
    className="hero"
  >
    <div
      className="copy-container"
    >
      <h2 />
      <h3>
        Weâ€™re going to change damn near everything about the way we furnish our living spaces.
      </h3>
    </div>
  </div>
  <div
    className="body-container"
  >
    <div
      className="copy-container"
    >
      <ReactMarkdown
        astPlugins={Array []}
        escapeHtml={true}
        plugins={Array []}
        rawSourcePos={false}
        renderers={Object {}}
        skipHtml={false}
        sourcePos={false}
        transformLinkUri={[Function]}
      />
      <HubspotForm
        formId="d1e13228-c396-4338-bbe3-67cd84f53065"
        portalId="4042167"
      />
    </div>
  </div>
</div>
`;
